.github/workflows/build-docker.yaml

```diff
@@ -0,0 +1,47 @@
+name: build and push Docker image
+
+on:
+  push:
+    tags:
+      - "release_[0-9]+_test[0-9]+"
+      - "release_[0-9]+"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+        with:
+          version: '270.0.0'
+          service_account_email: ${{ secrets.GCP_STG_EMAIL }}
+          service_account_key: ${{ secrets.GCP_STG_PRIVATE_KEY }}
+          export_default_credentials: true
+      - name: Configure GCR Docker auth (stg)
+        run: gcloud auth configure-docker
+      - name: Build docker image
+        run: docker build --tag ml-agents:${GITHUB_SHA} --build-arg SHA=${GITHUB_SHA} .
+      - name: Tag docker image (stg)
+        run: |
+          docker tag ml-agents:${GITHUB_SHA} gcr.io/unity-ai-ml-agents-stg/ml-agents:${GITHUB_SHA}
+          docker tag ml-agents:${GITHUB_SHA} gcr.io/unity-ai-ml-agents-stg/ml-agents:latest
+      - name: Push docker image (stg)
+        run: |
+          docker push gcr.io/unity-ai-ml-agents-stg/ml-agents:${GITHUB_SHA}
+          docker push gcr.io/unity-ai-ml-agents-stg/ml-agents:latest
+      - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
```

Comment:
Do you want the "prod" runs to execute on test tags? If not you can do something like this:
https://github.com/Unity-Technologies/ml-agents/blob/2dcfebdb49219702d82618dd955c25e26942cef3/.github/workflows/publish_pypi.yaml#L43

(I don't know if there's a better way to do this without repeating the commands and conditions)

Reply:
I think it would be OK to just not trigger this job at all except on full release tags.